mitosheet/mitosheet/utils.py

```diff
@@ -141,6 +143,32 @@ def dfs_to_array_for_json(
     return new_array
 
 
+def add_columns_to_condition_formats(
+    conditional_formats: Optional[List[Dict[str, Any]]],
+    df: any,
+    column_ids: ColumnIDMap,
+    sheet_index: int
+) -> Any:
+    if conditional_formats is None:
+        return None
+
+    export_cond_formats = []
+    for conditional_format in conditional_formats:
+        # Create new object to store the columns in the excel format
+        new_format = {
+            'columns': [],
+            'filters': conditional_format.get('filters'),
+            'font_color': conditional_format.get('color'),
+            'background_color': conditional_format.get('backgroundColor')
+        }
+        export_cond_formats.append(new_format)
+        for column_id in conditional_format.get('columnIDs'):
+            column_header = column_ids.get_column_header_by_id(sheet_index, column_id)
+            column_index = df.columns.tolist().index(column_header)
+            column = get_column_from_column_index(column_index)
+            new_format['columns'].append(column)
```

Comment:
Another option for the public interface: 
1. Make the `columns` passed to the `add_formatting_to_excel_sheet` function in the conditional format the actual column_headers rather than the excel column names
2. Pass the dataframe into the`add_formatting_to_excel_sheet` - and do this logic for finding the excel column inside that function. 

This would be nice because it:
1. Doesn't expose users to ColumnIDs.
2. Makes it really easy for users to edit the generated code if they want. - they wouldn't need to go backwards from Excel column -> their data column. 

What do you think?

Reply:
Ah yeah that definitely makes sense. Added!